python/tskit/tables.py

```diff
@@ -2924,20 +2945,44 @@ def drop_index(self):
         """
         self._ll_tables.drop_index()
 
-    def subset(self, nodes, record_provenance=True):
+    def subset(
+        self,
+        nodes,
+        record_provenance=True,
+        filter_populations=True,
+        filter_individuals=True,
+        filter_sites=True,
+        canonicalise=False,
+    ):
         """
         Modifies the tables in place to contain only the entries referring to
         the provided list of nodes, with nodes reordered according to the order
         they appear in the list. See :meth:`TreeSequence.subset` for a more
         detailed description.
 
+        Note: the tables can be completely unsorted.
+
         :param list nodes: The list of nodes for which to retain information. This
             may be a numpy array (or array-like) object (dtype=np.int32).
         :param bool record_provenance: Whether to record a provenance entry
             in the provenance table for this operation.
+        :param bool filter_populations: Whether to remove populations not referenced by
+            retained nodes. If False, the population table will remain unchanged.
```

Comment:
They will be reordered, though, right?

Reply:
No! I changed the behavior, to match `simplify`.